install/installer/pkg/config/v1/experimental/experimental.go

```diff
@@ -205,7 +205,8 @@ type ProxyConfig struct {
 }
 
 type PublicAPIConfig struct {
-	Enabled bool `json:"enabled"`
+	Enabled  bool  `json:"enabled"`
+	HttpPort int32 `json:"httpPort"`
```

Comment:
Why should it be configurable? I think we want it on 9002 always

Reply:
We want the public API to serve HTTP in Gitpod SaaS but not in self-hosted installations. If the `httpPort` setting is omitted, then the public API won't start the HTTP server (because `baseserver` won't start listening without a configured port).

Would you prefer the setting to be a boolean (`serveHTTP`)? Then we can hardcode the port to `9002` if it's set.

Reply:
Hmm, I think we should instead run the server and just have a No-OP implementation of the endpoint. Turning a whole HTTP server off in self-hosted is bound to bite us when we go to roll out. In practice. you'd need the same config on the proxy and other places and it just gets too complicated. It's better if it runs, just returns "Unvailable in Self-hosted".

Reply:
Not starting the HTTP server in self-hosted is cleaner as then there is less redundant, SaaS-only stuff floating around in a self-hosted instance.

I think the config we need is:
* This PR.
* Make the `proxy` config that we need to add to expose this HTTP endpoint optional (with another installer PR very similar to this one but for `proxy`).

I'm not sure it's inevitably going to go wrong at rollout, as we have good test coverage here with regards to the `yaml` rendered by the installer.

Reply:
Currently, the Public API server is in experimental and not deployed in Self-Hosted anyway. I'd avoid adding extra complexity at this stage which further changes which parts of the server are running and are not - it's already hard to understand what is running and what is not.

Basically, I'd drop the config for the HTTP port and always run the HTTP server on 9002

Reply:
> Basically, I'd drop the config for the HTTP port and always run the HTTP server on 9002

Was just typing the same comment - 💯 agree

Reply:
Ok, so I'll drop the new config surface here and always have the public api open for HTTP on port 9002, even in self-hosted. That means closing https://github.com/gitpod-io/ops/pull/3637 and https://github.com/gitpod-io/gitpod/pull/11833.

We can think about returning a no-op response for self hosted separately.

Do you agree @easyCZ ?